File: euler/geometry.py
def triangle_area(a: (int, int), b: (int, int), c: (int, int)) -> float:
    """Get area of triangle using coordinates of points

    Args:
        a, b, c: Tuple-like points where first element is coordinate by x and second by y

    Examples:
        >>> print(triangle_area((0, 0), (0, 3), (4, 0)))
        6.0
        >>> print(triangle_area((-23, 11), (19, 95), (-12, 114)))
        1701.0
    """
    return abs((a[0] - c[0]) * (b[1] - a[1]) - (a[0] - b[0]) * (c[1] - a[1])) / 2

File: euler/R/geometry.R
#' Get area of triangle using coordinates of points
#'
#' @param a: Point
#' @param b: Point
#' @param c: Point
#' @returns: Area of triangle
#' @examples
#' triangle_area(c(0, 0), c(0, 3), c(4, 0))
#' triangle_area(c(-23, 11), c(19, 95), c(-12, 114))
triangle_area <- function(a, b, c) abs((a[1] - c[1]) * (b[2] - a[2]) - (a[1] - b[1]) * (c[2] - a[2])) / 2
